dali/test/python/test_operator_roi_random_crop.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+import nvidia.dali as dali
+import nvidia.dali.fn as fn
+import nvidia.dali.types as types
+import nvidia.dali.math as math
+from test_utils import check_batch, dali_type
+import random
+from nose.tools import assert_raises
+
+np.random.seed(4321)
+
+def random_shape(min_sh, max_sh, ndim):
+  return np.array(
+      [np.random.randint(min_sh, max_sh) for s in range(ndim)],
+      dtype=np.int32
+    )
+
+def batch_gen(max_batch_size, sample_shape_fn, dtype=np.float32):
+    bs = np.random.randint(1, max_batch_size)
+    data = []
+    for i in range(bs):
+        sample_sh = sample_shape_fn()
+        data += [np.zeros(sample_sh, dtype=dtype)]
+    return data
+
+def check_roi_random_crop(ndim=2, max_batch_size=16,
+                          roi_min_start = 0, roi_max_start = 100,
+                          roi_min_extent = 20, roi_max_extent = 50,
+                          crop_min_extent = 20, crop_max_extent = 50,
+                          in_shape_min = 400, in_shape_max = 500,
+                          niter=3):
+    pipe = dali.pipeline.Pipeline(batch_size=max_batch_size, num_threads=4, device_id=0, seed=1234)
+    with pipe:
+        assert in_shape_min < in_shape_max
+        shape_gen_fn = lambda: random_shape(in_shape_min, in_shape_max, ndim)
+        data_gen_f = lambda: batch_gen(max_batch_size, shape_gen_fn)
+        shape_like_in = dali.fn.external_source(data_gen_f, device='cpu')
+        in_shape = dali.fn.shapes(shape_like_in, dtype=types.INT32)
+
+        crop_shape =  [(crop_min_extent + crop_max_extent) // 2] * ndim if random.choice([True, False]) \
```

Comment:
Numpy arrays are passed as argument inputs, so this still does not test non-tensor arguments. It must be a plain list.

Reply:
You are right. I fixed that now